recipes-app/iot2050-firmware-update/files/iot2050-firmware-update.tmpl

```diff
@@ -465,28 +472,32 @@ class ManagedFirmwareUpdate(FirmwareUpdate):
         # extract file path
         self.extract_path = "/tmp"
 
-    def update_firmware(self):
         # Get available firmwares by checking the board name and the os information
-        firmware_names = self._update_conf.firmware_filter(
+        self.firmware_names = self._update_conf.firmware_filter(
             self._board_info.board_name, self._board_info.os_info)
 
-        if len(firmware_names) <= 0:
-            raise UpgradeError("No firmware image could be updated on this board")
+    def check_firmware(self):
+        if len(self.firmware_names) <= 0:
+            raise UpgradeError("OS image version must be newer than the "
+                "minimal version, no firmware image could be updated on this"
+                " board!")
```

Comment:
Word `board` is more developer-oriented, better to use something user more familiar with, such as `device`.